linux/just_files/just_makeself_functions.bsh

```diff
@@ -214,92 +214,79 @@ function makeself_defaultify()
     #     ;;
     #
     # .. seealso::
-    #   :cmd:`makeself_just-project`
+    #   :cmd:`makeself_just-project` :cmd:`makeself_add-git-files`
     #**
     makeself_add-files) # Append files to a makeself executable
-      export MAKESELF_SOURCE_DIR="${1}"
       if [ "${2+set}" ]; then
         extra_args=2
       else
         extra_args=1
       fi
-      shift 1
 
-      _makeself_docker_compose run makeself add-files ${@+"${@}"}
+      _makeself_docker_compose run makeself add-files "${1}" ${2+"${2}"}
       ;;
+    #**
+    # .. command:: makeself_add-git-files
+    #
+    # :Arguments: * ``$1`` - Directory to add (can be relative or absolute)
+    #             * ``[$2]`` - Extra tar flags to be passed to makeself/tar
+    #
+    # After the initial just executable is created by calling :command:`makeself_just-project`, git project directories are added using :command:`makeself_add-git-files`
+    #
+    # :cmd:`makeself_add-git-files` has many advantages over :cmd:`makeself_add-files`
+    #
+    # #. No need to exclude .git, this is added to ``TAR_EXTRA`` for you
+    # #. No need to setup the path transform, this is auto determined for you
+    # #. Only git tracked files are added. Ignored and untracked files are not added. This both simplifies the exclude pattern and prevents accidental including of large files.
+    #
+    # The second argument can still be used to exclude files or pass any other arguments to ``tar``.
+    #
+    # .. rubric:: Example just targets
+    #
+    # .. code-block:: bash
+    #
+    #   makeself) # Simple case
```

Comment:
I think they are best served where they are, since the examples for `makeself_add-files` and makeself_add-git-files` are so different. `makeself_add-git-files` _should_ be easier to use.

There is no example for `makeself_just-project`